src/components/ui/UiPanelControlButton.vue

```diff
@@ -0,0 +1,70 @@
+<script lang="ts" setup>
+import { computed, type PropType } from 'vue'
+
+import arrowTopLine from '@/assets/icons/arrow-top-line.svg'
+import arrowRightLine from '@/assets/icons/arrow-right-line.svg'
+import arrowBottomLine from '@/assets/icons/arrow-bottom-line.svg'
+import arrowLeftLine from '@/assets/icons/arrow-left-line.svg'
+
+type AnchorPosition = 'top' | 'right' | 'bottom' | 'left'
+
+const props = defineProps({
+  isOpen: {
+    type: Boolean,
+    default: true,
+  },
+  anchorPosition: {
+    type: String as PropType<AnchorPosition>,
+    required: true,
+  },
+})
+
+const arrowIcon = computed(() => {
+  switch (props.anchorPosition) {
+    case 'top': {
+      return props.isOpen ? arrowTopLine : arrowBottomLine
+    }
+    case 'right': {
+      return props.isOpen ? arrowRightLine : arrowLeftLine
+    }
+    case 'bottom': {
+      return props.isOpen ? arrowBottomLine : arrowTopLine
+    }
+    case 'left': {
+      return props.isOpen ? arrowLeftLine : arrowRightLine
+    }
+    // Shouldn't happen
+    default:
+      return ''
+  }
+})
+
+const styleClass = computed(() => {
+  switch (props.anchorPosition) {
+    case 'top': {
+      return 'rounded-b-xl w-12 h-8 py-2 px-4'
+    }
+    case 'right': {
+      return 'rounded-l-xl w-8 h-12 py-4 px-2'
+    }
+    case 'bottom': {
+      return 'rounded-t-xl w-12 h-8 py-2 px-4'
+    }
+    case 'left': {
+      return 'rounded-r-xl w-8 h-12 py-4 px-2'
+    }
+    // Shouldn't happen
+    default:
+      return ''
+  }
+})
+</script>
+
+<template>
+  <button
+    class="flex justify-center items-center gap-4 bg-white shadow"
+    :class="styleClass"
+  >
+    <img :src="arrowIcon" />
```

Comment:
Yeah, this bothers me too actually.
In the beginning, I want to use it as a direct SVG because I can see the file directly and what it looks like, and easy to replace the image. But, over time there are some cases where I need to modify the SVG, so I need to make a component of it. The drawback is I can't see what it looks like.

I am open to any approach. We can decide if we want to create a component for all SVG or only create the component when we need it. Let me know what you think.